src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation/build/net461/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.targets

```diff
@@ -0,0 +1,47 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$(MSBuildThisFileDirectory)..\common.targets" />
+  <Target
+      Name="MvcRazorPrecompile"
+      DependsOnTargets="_ResolveInputArguments"
+      Inputs="$(MSBuildThisFileFullPath);@(MvcRazorFilesToCompile);@(IntermediateAssembly);@(DocFileItem);@(_DebugSymbolsIntermediatePath);@(ReferencePath);$(MSBuildAllProjects);"
+      Outputs="$(_MvcRazorOutputFullPath)">
+
+    <PropertyGroup Condition="'$(MvcRazorRunCommand)'==''">
+      <MvcRazorRunCommand Condition="'$(PlatformTarget)'=='x86'">$(OutputPath)$(MSBuildThisFileName)-x86.exe</MvcRazorRunCommand>
+      <MvcRazorRunCommand Condition="'$(PlatformTarget)'!='x86'">$(OutputPath)$(MSBuildThisFileName).exe</MvcRazorRunCommand>
+    </PropertyGroup>
+
+    <CallTarget Targets="_CreateResponseFileForMvcRazorPrecompile" />
+
+    <ItemGroup Condition="'$(PlatformTarget)'=='x86'">
+      <FilesToCopy Include="$(OutputPath)$(AssemblyName).exe.config">
```

Comment:
`FilesToCopy` is a fairly generic itemgroup name. My concern is that other targets may populate items into this. If they do, the `Copy` and `Delete` tasks below may fail or operation on the wrong files.

To be safe, give it a more specific name (`_PreCompilationFilesToCopy`) or clear it before using it `<FilesToCopy Remove="@(FilesToCopy)" />`.